### PR TITLE
planning: lattice_planner_dev branch merged into master branch

### PR DESCRIPTION
**Pull request by YajiaZhang**

### CODE REVIEW

File: modules/planning/lattice/behavior_decider/adc_master_scenario.cc [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/***************************************************************************
+ *
```
Comment (ycool) [inline modules/planning/lattice/behavior_decider/adc_master_scenario.cc]: use apollo common header template

File: modules/planning/lattice/behavior_decider/adc_master_scenario.cc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Copyright (c) 2016 Baidu.com.0, Inc. All Rights Reserved
+ *
+ **************************************************************************/
+/**
+ * @file adc_master_scenario.cpp
```
Comment (ycool) [inline modules/planning/lattice/behavior_decider/adc_master_scenario.cc]: remove adc_master_scenario.cpp. @file can auto generate the file name in doxgen

File: modules/planning/lattice/behavior_decider/scenario_manager.cc [hunk 3]
```diff
@@ -0,0 +1,72 @@
+/***************************************************************************
```
Comment (ycool) [inline modules/planning/lattice/behavior_decider/scenario_manager.cc]: fix comment header

File: modules/planning/lattice/behavior_decider/signal_light_scenario.cc [hunk 4]
```diff
@@ -0,0 +1,180 @@
+/***************************************************************************
```
Comment (ycool) [inline modules/planning/lattice/behavior_decider/signal_light_scenario.cc]: fix comment header

File: modules/planning/lattice/behavior_decider/signal_light_scenario.h [hunk 5]
```diff
@@ -0,0 +1,58 @@
+/***************************************************************************
```
Comment (ycool) [inline modules/planning/lattice/behavior_decider/signal_light_scenario.h]: fix comment header

File: modules/planning/lattice/trajectory_generator/end_condition_sampler.cc [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  }
+  return end_s_conditions;
+}
+
+std::vector<std::pair<std::array<double, 3>, double>>
+EndConditionSampler::SampleLonEndConditionsForPathTimeBounds(
+    const PlanningTarget& planning_target) const {
+  std::vector<std::pair<std::array<double, 3>, double>> end_s_conditions;
+
+  /**
```
Comment (ycool) [inline modules/planning/lattice/trajectory_generator/end_condition_sampler.cc]: remove these big comment out code block?

File: modules/planning/lattice/trajectory_generator/trajectory1d_generator.cc [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+Trajectory1dGenerator::~Trajectory1dGenerator() {
+  delete end_condition_sampler_;
+}
+
+void Trajectory1dGenerator::GenerateTrajectoryBundles(
+    const PlanningTarget& planning_target,
+    std::vector<std::shared_ptr<Curve1d>>* ptr_lon_trajectory_bundle,
+    std::vector<std::shared_ptr<Curve1d>>* ptr_lat_trajectory_bundle) {
+  /**
```
Comment (ycool) [inline modules/planning/lattice/trajectory_generator/trajectory1d_generator.cc]: remove these big comment out block. or add gflag to protect it

File: modules/planning/lattice/util/lattice_constraint_checker.cc [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    if (!fuzzy_within(a, FLAGS_longitudinal_acceleration_lower_bound,
+                      FLAGS_longitudinal_acceleration_upper_bound)) {
+      // AINFO << "not valid longitudinal trajectory: acceleration\t" << a;
+      return false;
+    }
+
+    double j = lon_trajectory.Evaluate(3, t);
+    if (!fuzzy_within(j, FLAGS_longitudinal_jerk_lower_bound,
+                      FLAGS_longitudinal_jerk_upper_bound)) {
+      // AINFO << "not valid longitudinal trajectory: jerk\t" << j;
```
Comment (ycool) [inline modules/planning/lattice/util/lattice_constraint_checker.cc]: remove these comment out debug log

File: modules/planning/lattice/util/lattice_constraint_checker.cc [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    double d2s_dt2 = lon_trajectory.Evaluate(2, t);
+
+    double a = 0.0;
+    if (s < lat_trajectory.ParamLength()) {
+      a = d2d_ds2 * ds_dt * ds_dt + dd_ds * d2s_dt2;
+    }
+
+    if (!fuzzy_within(a, -FLAGS_lateral_acceleration_bound,
+                      FLAGS_lateral_acceleration_bound)) {
+      // std::cout << "not valid lateral trajectory: acceleration\t" << a <<
```
Comment (ycool) [inline modules/planning/lattice/util/lattice_constraint_checker.cc]: remove cout debug

File: modules/planning/lattice/behavior_decider/adc_master_scenario.cc (hunk 1 above)
Comment (YajiaZhang) [reply modules/planning/lattice/behavior_decider/adc_master_scenario.cc]: Fixed the comment styles and removed outdated print messages.

File: modules/planning/lattice/behavior_decider/behavior_decider.h [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file collision_checker.h
```
Comment (Capri2014) [inline modules/planning/lattice/behavior_decider/behavior_decider.h]: collision checker ?

Comment (YajiaZhang) [reply modules/planning/lattice/behavior_decider/behavior_decider.h]: Will be fixed in next commit.

File: modules/planning/lattice/behavior_decider/condition_filter.cc [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+
+  return timestamps;
+}
+
+// Compute pixel img for lattice st
+bool ConditionFilter::GenerateLatticeStPixels(
+  apollo::planning_internal::LatticeStTraining* st_data,
+  double timestamp,
+  std::string st_img_name) {
+  int num_rows = 250;
```
Comment (Capri2014) [inline modules/planning/lattice/behavior_decider/condition_filter.cc]: I would suggest moving those into configuration as well in next iteration.

File: modules/planning/lattice/behavior_decider/scenario_manager.cc [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+namespace apollo {
+namespace planning {
+
+ScenarioManager::ScenarioManager() {}
+
+void ScenarioManager::RegisterScenarios() {
+  scenarios_.clear();
+  scenarios_.resize(NUM_LEVELS);
+  // level 0 features
```
Comment (Capri2014) [inline modules/planning/lattice/behavior_decider/scenario_manager.cc]: We only have level 0 features as of now?

File: modules/planning/lattice/trajectory_generator/trajectory_evaluator.cc [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+    const PlanningTarget& planning_target,
+    const std::shared_ptr<Trajectory1d>& lon_trajectory,
+    const std::shared_ptr<Trajectory1d>& lat_trajectory,
+    std::vector<double>* cost_components) const {
+  // currently consider three costs:
+  // 1. the cost of jerk, currently only consider longitudinal jerk.
+  // 2. the cost of time to achieve the objective
+  // 3. the cost of failure to achieve the planning objective.
+
+  double weight_lon_travel = 1000.0;
```
Comment (Capri2014) [inline modules/planning/lattice/trajectory_generator/trajectory_evaluator.cc]: So does the weight, maybe put them in param configs in next iteration

Comment (kechxu) [reply modules/planning/lattice/trajectory_generator/trajectory_evaluator.cc]: Yes, we will move the constants to gflags in next iterations.

File: modules/planning/lattice/behavior_decider/behavior_decider.cc [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file behavior_decider.cc
```
Comment (lianglia-apollo) [inline modules/planning/lattice/behavior_decider/behavior_decider.cc]: remove file name

**Pull request merged**